src/lib/modules/contracts_manager/index.js

```diff
@@ -226,6 +227,11 @@ class ContractsManager {
 
     if(resetContracts) self.contracts = {};
     async.waterfall([
+      function beforeBuild(callback) {
+        self.plugins.emitAndRunActionsForEvent("build:beforeAll", () => {
```

Comment:
Just curious, that's just something that might be needed in the future?

Reply:
see https://github.com/embark-framework/embark-snark/blob/master/index.js#L150

It is used by snark plugin